rotating_proxies/expire.py

```diff
@@ -32,6 +36,11 @@ class Proxies(object):
     """
     def __init__(self, proxy_list, backoff=None):
         self.proxies = {url: ProxyState() for url in proxy_list}
+        self.proxies_by_hostport = {}
+        for proxy in proxy_list:
+            parsed_proxy = _parse_proxy(proxy)
+            self.proxies_by_hostport[parsed_proxy[3]] = proxy
```

Comment:
could you please create a function which does `_parse_proxy(proxy)[3]` and add tests for it (doctests should be fine)? I think it'll allow to save a bit of code lines (this logic is repeated in get_proxy method), and make code a bit more clear.